buildbot/configure.py

```diff
@@ -103,10 +91,28 @@ def do_configure(args):
         llvm_build_shared_libs = 'ON'
 
     if args.use_lld:
-      llvm_enable_lld = 'ON'
+        llvm_enable_lld = 'ON'
 
-    if build_libclc:
-        llvm_enable_projects += ';libclc'
+    # CI Default conditionally appends to options, keep it at the bottom of
+    # args handling
+    if args.ci_defaults:
+        print("#############################################")
+        print("# Default CI configuration will be applied. #")
+        print("#############################################")
+
+        # For clang-format and clang-tidy
+        llvm_enable_projects += ";clang-tools-extra"
+        # libclc is required for CI validation
+        if 'libclc' not in llvm_enable_projects:
```

Comment:
It looks like we have code duplication now for:
 - --c--defaults
 - --cuda
Most of the parameters are set similary. Can we have one place for storing this data both for --cuda and --ci-defaults?

Reply:
The way I saw it was that part of the point of `ci-defaults` is that it needs to check what was used in the process so far and append to it, so it's quite hard to find a good common part.

I see your point re duplication, tried to get rid of those target names used in multiple places here: 263dafbe0f3a152521497631c4b7ddbac96be1b9